### ISSUE TITLE
Forging Countdown Clock Does Not Appear

### ISSUE THREAD

**Issue by brianlong** [incentivetoken/offspring#1]
The countdown clock for forging is working on one of my computers, but not another. Is there a setting that controls when the countdown clock appears? Thanks.

(Mac OS X 10.9.2 on both machines. Java 1.7.0. Only running on one machine at a time.)

**Comment by incentivetoken** [comment]
Forging is only possible for accounts with a non zero balance and that have that non zero balance for at least 1440 blocks. The blockchain should also be fully scanned and the balance above the forging checkbox has to be non-zero. 

Forging can be turned and off by checking/unchecking the checkbox beneath your account balance.

Did you try to turn on and off forging with the checkbox next to the countdown clock or turn Offspring on and off again?

If all fails you can always reset Offspring/NXT from the main menu. Go to File / Delete Blockchain to 'reset' the blockchain. This will do a soft reset. Offspring will clear out the entire blockchain (well actually NXT does it under the hood) which is stored in the a database.

If however even the blockchain database gets corrupted then you could delete the database all together and when Offspring starts a new database will be constructed and the blockchain will be downloaded.

You can find the database in the Offspring folder in your HOME directory. The database is the entire folder called 'nxt_db' or 'nxt_test_db'. So you'd delete the entire folder.

I hope this fixes your problem.

**Comment by brianlong** [comment]
I have a positive balance from transactions on March 4 & 7. They are both confirmed in the blockchain.

I tried the soft reset and a complete reset (delete folder, copy wallet.dat into new folder, start Offspring).

The forging box is checked, but the countdown clock does not appear.

It looks like Offspring is processing the blocks OK. Here is the output of the log (with debug=true):

[2014-03-28 07:55:02.071] Genesis block already in database
[2014-03-28 07:55:02.071] Scanning blockchain...
[2014-03-28 07:55:03.109] DEBUG: processed block 0
[2014-03-28 07:55:03.761] DEBUG: processed block 5000
[2014-03-28 07:55:04.126] DEBUG: processed block 10000
[2014-03-28 07:55:04.446] DEBUG: processed block 15000
[2014-03-28 07:55:04.743] DEBUG: processed block 20000
[2014-03-28 07:55:06.736] DEBUG: processed block 25000
[2014-03-28 07:55:07.736] DEBUG: processed block 30000
[2014-03-28 07:55:08.516] DEBUG: processed block 35000
[2014-03-28 07:55:09.781] DEBUG: processed block 40000
[2014-03-28 07:55:10.342] DEBUG: processed block 45000
[2014-03-28 07:55:11.021] DEBUG: processed block 50000
[2014-03-28 07:55:11.480] DEBUG: processed block 55000
[2014-03-28 07:55:11.897] DEBUG: processed block 60000
[2014-03-28 07:55:12.276] DEBUG: processed block 65000
[2014-03-28 07:55:12.721] DEBUG: processed block 70000
[2014-03-28 07:55:13.108] DEBUG: processed block 75000
[2014-03-28 07:55:13.460] DEBUG: processed block 80000
[2014-03-28 07:55:13.830] DEBUG: processed block 85000
[2014-03-28 07:55:14.154] DEBUG: processed block 90000
[2014-03-28 07:55:14.538] DEBUG: processed block 95000
[2014-03-28 07:55:15.134] ...done
[2014-03-28 07:55:15.175] Started peer networking server at 0.0.0.0:7874
[2014-03-28 07:55:15.176] DEBUG: Starting 8 background jobs
[2014-03-28 07:55:15.178] DEBUG: Initialization took 13 seconds
[2014-03-28 07:55:15.178] Nxt server 0.8.11 started successfully.
[2014-03-28 07:55:15.237] DEBUG: Account 13361823641112216679 started forging, deadline 16918925 seconds

**Comment by incentivetoken** [comment]
And account 13361823641112216679 is the account you expect to start forging? Or does that account display the countdown clock correct?

**Comment by brianlong** [comment]
Yes. That is the only account in the wallet. FWIW, the BTC and EUR balances are both 0.00 too. The NXT balance is 2247.

**Comment by brianlong** [comment]
Screen shot:

![screen shot 2014-03-28 at 8 55 16 am](https://cloud.githubusercontent.com/assets/79186/2550973/0dd7d0d4-b689-11e3-8e4f-e026d496830c.png)

**Comment by incentivetoken** [comment]
The fact that your BTC and EUR are not showing is a bug. You could try to click on "account-001" to re-focus that account. That will refresh the UI and might pick up those balances.

Regarding the not showing countdown clock. Not sure if that is on Offspring. The refreshing of that data goes through a completely different code path then the BTC and EUR labels. I tested it out over here and it looks like accounts with a very small balance simply do not get a remaining time reported by NXT maybe since it's to far in the future. The log however tells the truth that the account is indeed forging. The chance to generate a block with such a stake is sadly very small.

**Comment by brianlong** [comment]
Thanks for the follow-up. From a UX perspective, a short message to indicate a very long forging time will be helpful. e.g. "> 1 month" or "> 1 year"

**Comment by incentivetoken** [comment]
Good point. I'll leave this bug open.
